### DISCUSSION TITLE
Custom selection format

### DISCUSSION THREAD

**Discussion by martinbarilik** [orchidjs/tom-select#116] [Ideas]
Hi, i am usually using selectboxes with options that have 200 > characters but the view port space is precious. I want to be able to limit the size of the selectboxes whenever i need. 

For example 

https://jsfiddle.net/mArtinko5MB/cyzLxq20/11/

I want to cut the selected text ( with appending 3 dots f.e ) and on mouse hover to show title with full text of option ( so user can see what is the option about if there is some critical text in the end and is cut out )

Any plans to do it ? 

With select2 i used templateSelection option described here https://select2.org/selections. In this case i was able to define selected text that differs from texts of each options in dropdown menu. And whenever the user selected an option he saw just the part of it's text, but on hover he could see it all.

**Comment by oyejorge** [comment]
> I want to cut the selected text ( with appending 3 dots f.e ) and on mouse hover to show title with full text of option ( so user can see what is the option about if there is some critical text in the end and is cut out )

To change the display on mouse hover, CSS is your best bet. For example something like this: https://jsfiddle.net/m5hkqL04/

> With select2 i used templateSelection option described here https://select2.org/selections. In this case i was able to define selected text that differs from texts of each options in dropdown menu. And whenever the user selected an option he saw just the part of it's text, but on hover he could see it all.

We have a number of render template options. See https://tom-select.js.org/docs/#rendering
